### ISSUE TITLE
Duplicate contract names found for

### ISSUE THREAD

**Issue by rrodriguezreche** [trufflesuite/truffle#3403]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Wrong warning messages when using `truffle compile` with truffle 5.1.46. Working correctly with truffle 5.1.10.

## Steps to Reproduce

Install the openzeppelin contracts NPM package and use them.

## Expected Behavior

No warnings or errors

## Actual Results

Got the following log:

```
mad@mad:~/geodb/rewards-emitter/ethereum$ truffle compile

Compiling your contracts...
===========================
> Compiling ./contracts/BatchTransfer.sol
> Compiling ./contracts/EmissionCurve.sol
> Compiling ./contracts/Emitter.sol
> Compiling ./contracts/EmitterBase.sol
> Compiling ./contracts/EmitterLocalDev.sol
> Compiling ./contracts/GeoToken.sol
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/interfaces/IGeoToken.sol
> Compiling ./contracts/utils/Pausable.sol
> Compiling @openzeppelin/contracts/GSN/Context.sol
> Compiling @openzeppelin/contracts/GSN/GSNRecipient.sol
> Compiling @openzeppelin/contracts/GSN/IRelayHub.sol
> Compiling @openzeppelin/contracts/GSN/IRelayRecipient.sol
> Compiling @openzeppelin/contracts/access/Roles.sol
> Compiling @openzeppelin/contracts/access/roles/MinterRole.sol
> Compiling @openzeppelin/contracts/introspection/IERC1820Registry.sol
> Compiling @openzeppelin/contracts/ownership/Ownable.sol
> Compiling @openzeppelin/contracts/token/ERC777/IERC777.sol
> Compiling @openzeppelin/contracts/token/ERC777/IERC777Recipient.sol
> Compiling @openzeppelin/contracts/token/ERC777/IERC777Sender.sol
> Artifacts written to /home/mad/geodb/rewards-emitter/ethereum/build/contracts
> Compiled successfully using:
   - solc: 0.5.7+commit.6da8b019.Emscripten.clang


> Duplicate contract names found for Pausable.
> This can cause errors and unknown behavior. Please rename one of your contracts.

> Duplicate contract names found for IERC1820Registry.
> This can cause errors and unknown behavior. Please rename one of your contracts.

> Duplicate contract names found for Ownable.
> This can cause errors and unknown behavior. Please rename one of your contracts.

> Duplicate contract names found for IERC777.
> This can cause errors and unknown behavior. Please rename one of your contracts.

> Duplicate contract names found for IERC777Recipient.
> This can cause errors and unknown behavior. Please rename one of your contracts.

> Duplicate contract names found for IERC777Sender.
> This can cause errors and unknown behavior. Please rename one of your contracts.
```

## Environment

* Operating System: Ubuntu 18.04
* Ethereum client: Web3.js v1.2.1
* Truffle version (`truffle version`): 5.1.46
* node version (`node --version`): v10.22.0
* npm version (`npm --version`): 6.14.6

**Comment by eggplantzzz** [comment]
Is it possible that the warnings are correct? You might have multiple libraries that define contracts of those names. 

Those messages were added a couple of patches ago and you will most likely experience the same problem as you will in 5.1.46. Truffle currently keeps track of contracts by name and each contract name must be unique. We will change this at some point in the future as this is not the best way to do this. If you have multiple contracts of the same name, one will probably clobber the other and result in unpredictable behavior that is not desired.

**Comment by rrodriguezreche** [comment]
@eggplantzzz ok, found the duplication. I have one big ass flattened .sol file where multiple contract names are defined, it is an old file from another project that also inherited from Openzeppelin. TIL that truffle takes all the contracts in the file and compiles them separately.

Closing this, thank you!
